src/mol-canvas3d/controls/trackball.ts

```diff
@@ -434,11 +452,38 @@ namespace TrackballControls {
 
         const _spinSpeed = Vec2.create(0.005, 0);
         function spin(deltaT: number) {
-            if (p.spinSpeed === 0) return;
+            if (p.animate.name !== 'spin' || p.animate.params.speed === 0 || _isInteracting) return;
 
-            const frameSpeed = (p.spinSpeed || 0) / 1000;
+            const frameSpeed = p.animate.params.speed / 1000;
             _spinSpeed[0] = 60 * Math.min(Math.abs(deltaT), 1000 / 8) / 1000 * frameSpeed;
-            if (!_isInteracting) Vec2.add(_rotCurr, _rotPrev, _spinSpeed);
+            Vec2.add(_rotCurr, _rotPrev, _spinSpeed);
+        }
+
+        let _rockAngleSum = 0;
+        let _rockDirection = 1;
+        const _rockSpeed = Vec2.create(0.005, 0);
+        function rock(deltaT: number) {
+            if (p.animate.name !== 'rock' || p.animate.params.speed === 0 || _isInteracting) return;
+
+            // TODO get rid of the 3.3 factor
```

Comment:
@dsehnal do you have a better idea for formulating this? I couldn't figure out how smoothly change direction at a specific angle, hence the magic factor.

Relatedly, could you look into how to add a ``PluginStateAnimation`` for this? To make rocking videos.

Reply:
The magic factor: maybe instead of ``_rockAngleSum`` animate a ``t \in [0, 1]``  parameter and make the ``currentAngle \in [-angle, +angle]`` a function of that?

-----

### Animations

Expose ``animation`` properties of the trackball. Something along the lines of:

```ts
{
  animations: {
    rock: { reset: () => ..., tick: t => ... },
    spin: ...
  }
}
```

Then in the ``PluginStateAnimation.setup`` call ``reset``` and ``tick`` in ``PluginStateAnimation.apply``. Can also remember the old state in ``setup`` and restore it in ``teardown``.

Reply:
> The magic factor: maybe instead of _rockAngleSum animate a t \in [0, 1] parameter and make the currentAngle \in [-angle, +angle] a function of that?

yeah, exactly what I could not figure out 😅